x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/ClusterPermission.java

```diff
@@ -90,11 +93,10 @@ public Builder add(final ClusterPrivilege clusterPrivilege, final Set<String> al
             return this;
         }
 
-        public Builder add(final ConfigurableClusterPrivilege configurableClusterPrivilege, final Predicate<String> actionPredicate,
-                           final Predicate<TransportRequest> requestPredicate) {
-            return add(configurableClusterPrivilege, new ActionRequestPredicatePermissionCheck(configurableClusterPrivilege,
-                                                                                               actionPredicate,
-                                                                                               requestPredicate));
+        public Builder add(final ClusterPrivilege clusterPrivilege, final Predicate<String> actionPredicate,
+                           final BiPredicate<TransportRequest, Authentication> requestAuthnPredicate) {
```

Comment:
I think it's simpler to use two `Predicate`s instead of one `BiPredicate`. Is there a reason for it?
If there is a clear requirement, I think it's best to have another type of `PermissionCheck` than to merge predicates like this.

Reply:
I have the impression that we're overthinking it. In the future we might have all kinds of `PermissionChecks`, but currently the grouping of the request and authentication predicates seems arbitrary to me. For example, we might as well group action name and request predicates because they are more related to each other. After all this is just one builder method, and we can always add others for every situation, but this still strikes me as an odd choice.